src/T2DMap.cpp

```diff
@@ -2423,18 +2420,16 @@ void T2DMap::createLabel(QRectF labelRectangle)
     float my2 = (yspan / 2.0) - (labelRectangle.bottomRight().y() / mRoomHeight) - mOy;
     label.pos = QVector3D(mx, my, mOz);
     label.size = QRectF(QPointF(mx, my), QPointF(mx2, my2)).normalized().size();
-    if (!mpMap->mpRoomDB->getArea(mAreaID)) {
+    auto pArea = mpMap->mpRoomDB->getArea(mAreaID);
+    if (!pArea) {
         return;
     }
-    int labelID;
-    if (!mpMap->mapLabels.contains(mAreaID)) {
-        QMap<int, TMapLabel> m;
-        m[0] = label;
-        mpMap->mapLabels[mAreaID] = m;
-    } else {
-        labelID = mpMap->createMapLabelID(mAreaID);
-        mpMap->mapLabels[mAreaID].insert(labelID, label);
-    }
+
+    int labelID = -1;
+    do {
+    } while (pArea->mMapLabels.contains(++labelID));
+    pArea->mMapLabels.insert(labelID, label);
+
```

Comment:
Same with the safety of `createMapLabelID`

Reply:
That safety is illusionary IMHO, but, given that it *is* needed in more than one place I guess it is okay to refactor the code out to it...